tests/spec/config_spec.lua

```
@@ -0,0 +1,301 @@
+      }
+    }
+    
+    config.setup(test_config)
+    local success = config.save()
+    
+    assert.is_true(success, "Should save configuration successfully")
+  end)
+  
+  it("should load configuration from file", function()
+    -- Mock file exists and contains config
+    vim.fn.filereadable = function(path) return 1 end
```

Luacheck (1.2.0)

[warning] mutating non-standard global variable 'vim'

(W112)